index.js

```diff
@@ -99,6 +99,19 @@ class HtmlWebpackPlugin {
       this.options.filename = path.relative(compiler.options.output.path, filename);
     }
 
+    const minify = this.options.minify;
+    if (minify === true || (minify === undefined && compiler.options.mode === 'production')) {
```

Comment:
Could you please add the following code to make it easier to understand?

```js
// Check if webpack is running in production mode 
// @see https://github.com/webpack/webpack/blob/3366421f1784c449f415cda5930a8e445086f688/lib/WebpackOptionsDefaulter.js#L12-L14
const isProductionLikeMode = compiler.options.mode === 'production' || !compiler.options.mode;
```

Reply:
Good idea :-)